### DISCUSSION TITLE
Using for loop for grouped plot

### DISCUSSION THREAD

**Discussion by sdasnoon** [datapane/datapane#112] [Help] [answered]
Hi, 

I am creating a report that will create plots based on unique values in one of the columns ( filters on the unique value of this column). The length of unique values in this column varies based on time, so if I take 3 months data, there might be 10 unique values, but if I take 6months data, there might be 15 unique values. 

### Is there a way to use dp.Group along with for loop. 

I am looking for something like this, the following codes are just for reference and just to provide a gist of the problem.
```
df= pd.DataFrame(Some_data_frame)
Unique_identifier =df['column_b'].unique()

all_plots =[]
```
> The below code picks two unique filter plots the graph for them and group them.

```
for i in range(0, len(Unique_identifier), 2):
      filtered_data_1 = df[df['column_b']==Unique_identifier[i]]
      plot_graph_1 = dp.plot(some_graph_1)
      filtered_data_2 = df[df['column_b']==Unique_identifier[i+1]]
      plot_graph_2 = dp.plot(some_graph_2)
      grouped_plot = dp.Group(plot_graph_1,plot_graph_2,columns=2)
      all_plot = all_plot.append(grouped_plot)

r= dp.Report(all_plots)
```

**Comment by mands** [comment, marked as answer]
Hi @sdasnoon ,

So your approach is generally correct, basically to handle dynamic / generated plots, the best approach is to add them to a list then pass that list into the Report.
The main issue here is that when using a list, you either need to destruct the list, using `*all_plots` or use the special `blocks` keyword argument that accepts lists of blocks directly.
For example,
```
r = dp.Report(*all_plots)
# or
r = dp.Report(blocks=all_plots)
```
will work as intended. Thanks!
